### PR TITLE
Test migration: All tests safari latest

### PR DESCRIPTION
**Pull request by metal-messiah**
### Overview 
**---Major changes PR---**

- Re-defines rudimentarily the browsers and versions cutoff that we support
- Updates to GH actions and JIL test suite
- Expand on last chrome@latest checkpoint to ensure **all** defined browser-version functional & unit test runs are passing
- Changes to some core source files and test files, especially SPA, to reflect working state

This is an all-passing checkpoint for the test migration project, barring cleanup and other post tasks.

### CODE REVIEW

File: packages/browser-agent-core/common/window/nreum.js [hunk 1]
```diff
@@ -94,6 +94,26 @@ export function gosNREUMInitializedAgents(id, obj, target) {
   return nr
 }
 
+function gosNRDEBUG() {
```
Comment (aubreymasten) [inline packages/browser-agent-core/common/window/nreum.js:97]: Previously this code was injected by JIL when tests were running. Is this temporary/will this be added to the production agent?

Comment (metal-messiah) [reply packages/browser-agent-core/common/window/nreum.js:97]: After thinking, I think this should actually be temporary

Comment (cwli24) [reply packages/browser-agent-core/common/window/nreum.js:97]: > After thinking, I think this should actually be temporary (and cleaned up)

Done in the cleanup branch; disregard this on here for now. Alternatively, can merge that in but not preferable as it'll bloat this one and make review harder for everyone to go through a bunch of commented-out removals.

**Comment by cwli24** [comment]
Please don't squash this when merging. Would "lose" the last few weeks of granular history on new `dev`, plus `main` is now tangled into this history.

**Pull request merged**